aten/src/THCUNN/generic/SpatialConvolutionMM.cu

```diff
@@ -6,18 +6,21 @@ static inline void THNN_(SpatialConvolutionMM_shapeCheck)(
                          THCState *state,
                          THCTensor *input, THCTensor *gradOutput,
                          THCTensor *weight, THCTensor *bias,
-                         int kH, int kW, int dH, int dW, int padH, int padW) {
+                         int kH, int kW, int dH, int dW, int padH, int padW,
+                         int weight_nullable) {
   THArgCheck(kW > 0 && kH > 0, 9,
              "kernel size should be greater than zero, but got kH: %d kW: %d", kH, kW);
   THArgCheck(dW > 0 && dH > 0, 11,
              "stride should be greater than zero, but got dH: %d dW: %d", dH, dW);
-  THArgCheck(!bias || THCTensor_(isContiguous)(state, bias), 5,
-             "bias tensor has to be contiguous");
-  THCUNN_argCheck(state, weight->nDimension == 2 || weight->nDimension == 4, 5, weight,
-                  "2D or 4D weight tensor expected, but got: %s");
 
-  if (bias != NULL) {
-    THCUNN_check_dim_size(state, bias, 1, 0, weight->size[0]);
+  if (weight != NULL) {
+    THCUNN_argCheck(state, weight->nDimension == 2 || weight->nDimension == 4, 5, weight,
+                    "2D or 4D weight tensor expected, but got: %s");
+    if (bias != NULL) {
+      THCUNN_check_dim_size(state, bias, 1, 0, weight->size[0]);
+    }
```

Comment:
Why can bias only be not null when weights are not null?

Reply:
Oh no that is not what I am trying to imply. This check uses `weight->size[0]` so it needs to be wrapped in `weight != NULL` case. It's a bit confusing since `weight` is really `gradWeight` here.